plugins/modules/packaging/os/openbsd_pkg.py

```diff
@@ -520,10 +534,13 @@ def main():
             name=dict(type='list', elements='str', required=True),
             state=dict(type='str', default='present', choices=['absent', 'installed', 'latest', 'present', 'removed']),
             build=dict(type='bool', default=False),
+            snapshot=dict(type='bool', default=False),
             ports_dir=dict(type='path', default='/usr/ports'),
             quick=dict(type='bool', default=False),
             clean=dict(type='bool', default=False),
         ),
+        mutually_exclusive=[['snapshot', 'name'], ['snapshot', 'state'], ['snapshot', 'quick'], ['snapshot', 'clean'],
+                            ['build', 'name'], ['build', 'state'], ['build', 'quick'], ['build', 'clean'], ['build', 'ports_dir']],
```

Comment:
Now you are also making combinations invalid that were available before. This could break existing playbooks/roles and should not be done without a deprecation.

(Also you didn't include `['build', 'snapshot']` in that list.)

Reply:
Thanks everyone for the continued work on this. Indeed I think converting to the mutually_exclusive list is the better approach, thanks for that. However the current list seems a bit strange, not being able to supply the name of the package together with either "build" or "snapshot" is obviously not what we want:
```
$ ansible -i hosts -u root openbsd_pkg_tests -m openbsd_pkg -a "name=vim--no_x11 snapshot=yes"
[WARNING]: You are running the development version of Ansible. You should only run Ansible from "devel" if you are modifying the Ansible engine, or trying out features under development. This is a rapidly changing source of
code and can become unstable at any point.
192.168.1.60 | FAILED! => {
    "changed": false,
    "msg": "parameters are mutually exclusive: snapshot|name"
}
```
I'm thinking what you want is this:
```
mutually_exclusive=[['build', 'snapshot']],
```
Also I don't see that we can remove this part which is currently being done:
```
if module.params['build'] is True:
        module.fail_json(msg="the combination of build=%s and state=latest is not supported" % module.params['build'])
```
It is only failing on specific setting of "state", it is not a mutually exclusive thing between "build" and "state" as a whole, so I am not sure why we are trying to remove it.

Reply:
Ops, I misunderstood "mutually exclusive", now it should fit. I have also re-added a modified "build=%s and state=latest" check.

Reply:
You haven't pushed anything yet.

Reply:
`s/master/main`, thanks for the hint